src/Dotnet.Script.DependencyModel/Context/ScriptDependencyContextReader.cs

```diff
@@ -0,0 +1,200 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Dotnet.Script.DependencyModel.Logging;
+using Dotnet.Script.DependencyModel.ScriptPackage;
+using Microsoft.DotNet.PlatformAbstractions;
+using NuGet.Common;
+using NuGet.Packaging;
+using NuGet.ProjectModel;
+using NuGet.RuntimeModel;
+
+namespace Dotnet.Script.DependencyModel.Context
+{
+    public class ScriptDependencyContextReader
+    {
+        private readonly Logger _logger;
+        private readonly ILogger _nuGetLogger;
+        private readonly ScriptFilesDependencyResolver _scriptFilesDependencyResolver;
+
+        public ScriptDependencyContextReader(LogFactory logFactory, ScriptFilesDependencyResolver scriptFilesDependencyResolver)
+        {
+            _logger = logFactory.CreateLogger<ScriptDependencyContextReader>();
+            _nuGetLogger = new NuGetLogger(logFactory);
+            _scriptFilesDependencyResolver = scriptFilesDependencyResolver;
+        }
+
+        public ScriptDependencyContextReader(LogFactory logFactory)
+        : this(logFactory, new ScriptFilesDependencyResolver(logFactory))
+        {
+        }
+
+        public ScriptDependencyContext ReadDependencyContext(string pathToAssetsFile)
+        {
+            var lockFile = LockFileUtilities.GetLockFile(pathToAssetsFile, _nuGetLogger);
+
+            // Since we execute "dotnet restore -r [rid]" we get two targets in the lock file.
+            // The second target is the one containing the runtime deps for the given RID.
+            var targetLibraries = lockFile.Targets[1].Libraries;
```

Comment:
can we add here some logic that checks the length first? I worry that in some edge cases this might starts throwing out of range exception

Reply:
Sure, no problem. I'll throw a meaningful exception here